### PR TITLE
Add support for gh-ost v1.1.6 command line flags and drop deprecated flag

### PR DESCRIPTION
**Pull request by nickdowse**
## Description

Add support for new command line flags in gh-ost [v1.1.5](https://github.com/github/gh-ost/releases/tag/v1.1.5) and [v1.1.6](https://github.com/github/gh-ost/releases/tag/v1.1.6):
* `--throttle-http-interval-millis`
* `--throttle-http-timeout-millis`
* `--attempt-instant-ddl`
* `--allow-zero-in-date`
* `--binlogsyncer-max-reconnect-attempts`

**Breaking Change** we're also removing support for `replication-lag-query` flag ahead of it's deprecation in `gh-ost` v1.1.7. 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines set by rubocop
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by giuanbalina — APPROVED** [review]

**Comment by epintos** [comment]
@nickdowse why is this needed? Are you using the newest version in frontend?

**Comment by epintos** [comment]
@nickdowse I'm not sure if it's worth investing time in stopping using this gem since support Rails 7.2 requires quite some refactors

**Comment by nickdowse** [comment]
@epintos In frontend we want to use instant DDL options (supported in `gh-ost` v1.1.6) now that we're on MySQL 8.0 - I believe this will reduce the issues we're seeing at table switchover.

File: CODEOWNERS
Comment (giuanbalina) [inline CODEOWNERS:1]: Maybe we can just remove the code owners

**Review by giuanbalina — APPROVED** [review]

**Pull request merged**